enhancements/node-latency-profile/node-latency-profiles.md

```diff
@@ -0,0 +1,280 @@
+
+MCO, KCMO and CKAO will have necessary observers to consume `NodeLatencyProfile`.
+
+You can also refer to this [flow diagram] below (https://raw.githubusercontent.com/harche/enhancements-1/reliability/enhancements/reliability/Reliability%20Profile%20Flow.png) for additional information (Please note that for the sake of brevity, the flow diagram doesn't highlight the rollback mechanism)
+
+![flow diagram](https://raw.githubusercontent.com/harche/enhancements-1/reliability/enhancements/reliability/Reliability%20Profile%20Flow.png)
+
+
+#### MCO
+
+1. The relavent observer in MCO watching `NodeLatencyProfile` will check for `NodeLatencyProfile.Status.KubeletUpdateStatus`. If it's set to `NotStartedState`, it means `NodeLatencyProfile` not been processed at all and indicating that this is a freshly created `NodeLatencyProfile`.
+2. MCO will try to read `NodeLatencyProfile.Spec.ProfileType`. Depending upon this `ProfileType`, it will assemble a new `KubeletConfig` object that has appropriate value for `nodeStatusUpdateFrequency` and update `NodeLatencyProfile.Status.KubeletUpdateStatus.KubeletConfigName` with the automatically generated name. Then the existing `KubeletConfigController` will be used to deploy this `KubeletConfig` to the all nodes of the cluster. MCO will update `NodeLatencyProfile.Status.KubeletUpdateStatus.State` to `InProgressState`.
+3. If the `KubeletConfigController` fails to reconcile that `KubeletConfig` the error will be propagated to the `NodeLatencyProfile.Status.KubeletUpdateStatus.Message` and the `NodeLatencyProfile.Status.KubeletUpdateStatus.State` will be set to `FailedState`.
+4. If on the other hand, `KubeletConfigController` successfully reconcils the given `KubeletConfig`, MCO will update `NodeLatencyProfile.Status.KubeletUpdateStatus.State` to `CompleteState`.
+5. Even though `KubeletConfig` is successfully reconciled, MCO observer will keep monitoring `NodeLatencyProfile.Status.ControllerManagerUpdateStatus.State`. If it gets set to `FailedState` or `RollBackState` which indicates that controller manager arguments update didn't go as planned and MCO observer will rollback Kubelet arguments by deleting the `KubeletConfig`.
+
+#### KCMO
+
+1. The relavent observer in KMCO watching `NodeLatencyProfile` wait until `NodeLatencyProfile.Status.KubeletUpdateStatus.State` to get set to `CompleteState`. This will ensure that the updates to Kubelet and Controller Manager are not carried out simultaneously.
+2. This KMCO observer will read `NodeLatencyProfile.Spec.ProfileType` and depending upon the `ProfileType` it will output appropriate value for the extended arguments `--node-monitor-period` and/or `--node-monitor-grace-period` of the controller manager.
+3. If KMCO is successful in updating the given arguments of the controller manager, it will update  `NodeLatencyProfile.Status.ControllerManagerUpdateStatus.State` to `CompleteState`.
```

Comment:
this is misplaced. kcm does not push back status, but only observes. If you want status about kcm/kube-apiserver, you have to do that from another place, e.g. MCO.